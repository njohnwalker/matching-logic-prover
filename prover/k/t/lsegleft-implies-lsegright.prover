claim \implies( \and( lsegleft (variable("H"), variable("X"), variable("Y"), variable("F"), .Patterns), .Patterns )
              , \and( lsegright(variable("H"), variable("X"), variable("Y"), variable("F"), .Patterns), .Patterns )
              )
strategy search-bound(5)

/*
strategy kt ; ( ( right-unfold ; direct-proof )
              | direct-proof
              | kt ; ( (right-unfold ; right-unfold ; direct-proof )
                     | direct-proof
                     | right-unfold ; direct-proof
                     )
              )
*/
