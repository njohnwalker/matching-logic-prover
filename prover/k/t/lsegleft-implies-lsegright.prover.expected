<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( lsegleft ( variable ( "H" ) , variable ( "X" ) , variable ( "Y" ) , variable ( "F" ) , .Patterns ) , .Patterns ) , \and ( lsegright ( variable ( "H" ) , variable ( "X" ) , variable ( "Y" ) , variable ( "F" ) , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
